### ISSUE TITLE
Make Policy Sentry endpoint configurable via Provider parameters

### ISSUE THREAD

**Issue by kmcquade** [reetasingh/terraform-provider-policy-sentry#6]
Since we are currently in the development stages, this will help us when we change the API gateway endpoint.

I got this error when running Terraform plan:

```
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

data.policy-sentry_document.example: Refreshing state...

Error: Post "https://zeok878mvj.execute-api.us-east-1.amazonaws.com/dev//write": dial tcp: lookup zeok878mvj.execute-api.us-east-1.amazonaws.com on 10.2.33.2:53: no such host
```

Example code:

```hcl
terraform {
  required_providers {
    policy-sentry = {
      source = "reetasingh/policy-sentry"
      version = "1.1.0"
    }
  }
}

provider "policy-sentry" {
  # Configuration options
}

provider "aws" {
    region = "us-east-1"
}

data "policy-sentry_document" "example" {
    read = list("arn:aws:s3:::mybucket")
}

# Returns policy sentry document in json
output "policy-sentry_document_json" {
  value = data.policy-sentry_document.example.json
}

resource "aws_iam_policy" "policy" {
  name        = "test-terraform-provider"
  path        = "/"
  description = "testing the terraform provider for Policy Sentry"
  policy      = data.policy-sentry_document.example.json
}
```

Desired provider configuration options:

```hcl
provider "policy_sentry" {
  # Once we get a long lived URL we can change it to that.
  endpoint = "https://zeok878mvj.execute-api.us-east-1.amazonaws.com/dev/write"
}
```

**Comment by reetasingh** [comment]
@kmcquade thanks for the issue.
I broke the code last night. you can resume your testing. It is fixed now.
Also I have added support for almost all input parameters
https://github.com/reetasingh/terraform-provider-policy-sentry/blob/main/policy_sentry/data_source_policy_sentry_document.go

**Comment by sahilvv** [comment]
https://github.com/reetasingh/terraform-provider-policy-sentry/pull/8

**Comment by kmcquade** [comment]
Closed by #8

**Comment by reetasingh** [comment]
@kmcquade  this is now fixed as part of v1.1.7. tested it by this
```
terraform {
  required_providers {
    policy-sentry = {
      source = "reetasingh/policy-sentry"
      version = "1.1.7"
    }
    aws = {
      source  = "hashicorp/aws"
      version = "~> 3.0"
    }
  }
}
# Configure the AWS Provider
provider "aws" {
  region = "us-east-1"
}
provider "policy-sentry" {
  endpoint = "<endpoint>"
}
data "policy-sentry_document" "example" {
    write = list("arn:aws:kms:us-east-1:123456789012:key/aaaa-bbbb-cccc")
    exclude_actions = list("s3:GetAccelerateConfiguration", "s3:GetAnalyticsConfiguration")
    read = list("arn:aws:s3:::mybucket")
    tagging = list("arn:aws:s3:::mybucket")
    permissions_management = list("arn:aws:s3:::mybucket")
}
# Returns policy sentry document in json
output "policy-sentry_document_json" {
  value = data.policy-sentry_document.example.json
}
resource "aws_iam_policy" "policy" {
  name        = "sample"
  path        = "/"
  description = "this uses policy sentry document"
  policy      = data.policy-sentry_document.example.json
}
